dvr-ws/routes/index.js

```diff
@@ -2,18 +2,34 @@ var express = require('express');
 var router = express.Router();
 var config = require('../../common/Configuration');
 var logger = require('../logger/logger');
-var qio = require('q-io/fs');
 var path = require('path')
 var persistenceFormat = require('../../common/PersistenceFormat');
+var fsUtils = require('../../lib/utils/fs-utils');
 
-router.get(/\/smil\:([^\\/]*)_all\.smil\/([^\?]*)/i, function(req, res) {
+// get chunklist expire age
+var chunklistExpireAge = config.get("webServerParams:chunklistExpireAge");
 
+router.get(/\/smil:([^\\/]*)_all\.smil\/([^\?]*)/i, function(req, res) {
     var entryId = req.params[0];
     var fileName = req.params[1];
 
-    var disk = path.join(persistenceFormat.getEntryFullPath(entryId),fileName);
-    res.sendFile(disk);
+    var fullPath = path.join(persistenceFormat.getEntryFullPath(entryId),fileName);
+
+    if ( fileName.search('chunklist.m3u8') > -1 ) {
+
+        fsUtils.checkIfFileExpired(fullPath, chunklistExpireAge)
+            .then(function () {
+                res.sendFile(fullPath);
+            })
+            .catch(function () {
+                res.status(404).send('File not found');
```

Comment:
1) streaming to single media server and dropping the stream the following error was returned (with 404, status), "<HTML><HEAD><TITLE>Error</TITLE></HEAD><BODY>
An error occurred while processing your request.<p>
Reference&#32;&#35;183&#46;d623f17&#46;1465802962&#46;53f7fa
</BODY></HTML>"
2) 2 streams, then drop one stream and then the other.
result: after dropping 1 stream the error message as above. After dropping the 2nd, player sometimes get ENDLIST tag in reply and sometimes 404 with above message. In either case, if both DC are unavailable the player stops. 
As agreed changing the error message to "File expired"